dev/list_changed_files.py

```diff
@@ -25,13 +26,14 @@ def main():
     per_page = 100
     page = 1
     token = os.environ.get("GITHUB_TOKEN")
-    headers = {"Authorization": f"token {token}"} if token else None
+    headers = {"Authorization": f"token {token}"} if token else {}
     # Ref: https://docs.github.com/en/rest/reference/pulls#list-pull-requests-files
     url = f"https://api.github.com/repos/{args.repository}/pulls/{args.pr_num}/files"
     while True:
-        resp = requests.get(url, params={"per_page": per_page, "page": page}, headers=headers)
-        resp.raise_for_status()
-        files = resp.json()
+        full_url = f"{url}?per_page={per_page}&page={page}"
+        req = urllib.request.Request(full_url, headers=headers)
+        resp = urllib.request.urlopen(req)
+        files = json.loads(resp.read().decode())
```

Comment:
We might want to use context manager on urlopen?
```suggestion
        with urllib.request.urlopen(req) as resp:
                files = json.loads(resp.read().decode())
```

Reply:
Great idea, updated!